### PR TITLE
Issues #5741 and #5742

### PR DESCRIPTION
**Pull request by nad**
Fixes for issues #5741 and #5742.

### CODE REVIEW

File: src/full/Agda/Compiler/CallCompiler.hs [hunk 1]
```diff
@@ -65,6 +71,9 @@ callCompiler' cmd args = do
     Just out -> forkTCM $ do
       -- The handle should be in text mode.
       liftIO $ hSetBinaryMode out False
+      case enc of
```
Comment (andreasabel) [inline src/full/Agda/Compiler/CallCompiler.hs:74]: Use `whenJust`.

File: src/full/Agda/Compiler/CallCompiler.hs [hunk 2]
```diff
@@ -73,6 +82,9 @@ callCompiler' cmd args = do
     Just err -> do
       -- The handle should be in text mode.
       hSetBinaryMode err False
+      case enc of
```
Comment (andreasabel) [inline src/full/Agda/Compiler/CallCompiler.hs:85]: Use `whenJust`.

File: src/full/Agda/Compiler/MAlonzo/Compiler.hs [hunk 3]
```diff
@@ -1290,8 +1292,10 @@ callGHC = do
 
   let ghcBin = optGhcBin opts
 
+  -- Make GHC use UTF-8 when writing to stdout and stderr.
+  liftIO $ setEnv "GHC_CHARENC" "UTF-8"
```
Comment (andreasabel) [inline src/full/Agda/Compiler/MAlonzo/Compiler.hs:1296]: Could this be moved to `callCompiler` to keep the simple API?
We could have `callCompiler'` for a more parametrized version of the API?

File: src/full/Agda/Utils/IO/UTF8.hs [hunk 4]
```diff
@@ -41,6 +42,15 @@ readTextFile file = convertLineEndings <$> do
   IO.hSetEncoding h IO.utf8
   T.hGetContents h
 
+-- | Reads a UTF8-encoded text file and converts many character
+-- sequences which may be interpreted as line or paragraph separators
+-- into '\n'.
+
+readFile :: FilePath -> IO String
+readFile f = do
+  s <- readTextFile f
```
Comment (andreasabel) [inline src/full/Agda/Utils/IO/UTF8.hs:51]: Use `<.>`: `readFile = T.unpack <.> readTextFile`.
Or at least use `<$>`.

**Review by andreasabel — COMMENTED** [review]
CI passes.

Some stylistic changes suggested.

File: src/full/Agda/Compiler/MAlonzo/Compiler.hs (hunk 3 above)
Comment (nad) [reply src/full/Agda/Compiler/MAlonzo/Compiler.hs:1296]: No, `callCompiler` is not only for calling GHC.

**Comment by nad** [comment]
@andreasabel, I don't want to spend time on adhering to your personal tastes right now, but I do not mind if you make the changes yourself.

**Pull request merged**